packages/core/src/modules/dids/methods/peer/__tests__/__fixtures__/didPeer2Ez6L.json

```diff
@@ -2,21 +2,21 @@
   "id": "did:peer:2.Ez6LSbysY2xFMRpGMhb7tFTLMpeuPRaqaWM1yECx2AtzE3KCc.Vz6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V.Vz6MkgoLTnTypo3tDRwCkZXSccTPHRLhF4ZnjhueYAFpEX6vg.SeyJ0IjoiZG0iLCJzIjoiaHR0cHM6Ly9leGFtcGxlLmNvbS9lbmRwb2ludCIsInIiOlsiZGlkOmV4YW1wbGU6c29tZW1lZGlhdG9yI3NvbWVrZXkiXSwiYSI6WyJkaWRjb21tL3YyIiwiZGlkY29tbS9haXAyO2Vudj1yZmM1ODciXX0",
   "authentication": [
     {
-      "id": "did:peer:2.Ez6LSbysY2xFMRpGMhb7tFTLMpeuPRaqaWM1yECx2AtzE3KCc.Vz6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V.Vz6MkgoLTnTypo3tDRwCkZXSccTPHRLhF4ZnjhueYAFpEX6vg.SeyJ0IjoiZG0iLCJzIjoiaHR0cHM6Ly9leGFtcGxlLmNvbS9lbmRwb2ludCIsInIiOlsiZGlkOmV4YW1wbGU6c29tZW1lZGlhdG9yI3NvbWVrZXkiXSwiYSI6WyJkaWRjb21tL3YyIiwiZGlkY29tbS9haXAyO2Vudj1yZmM1ODciXX0#6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V",
+      "id": "#key-2",
```

Comment:
It would be good to keep full ids here, at least in the 'resolved' variant as using relative identifiers is not valid and might lead to unexpected behavior

Reply:
I think for the resolution it is OK to keep them, as you say. For creation, [the spec](https://identity.foundation/peer-did-method-spec/#creating-a-did) now states: _"All identifiers within this document MUST be relative. For example, the id of a verificationMethod might be #key-1 instead of something like did:example:abc123#key-1."_

In did:peer:2, key encoding ordering is important, as the resolver will need to reconstruct relative references (e.g. encoded within services) using "key-N" pattern, which was the issue found by @dbluhm when testing against ACA-Py.

Reply:
I think this should say "All identifiers within _the encoded_ document MUST be relative". Otherwise it goes against the did-core specification, which mentions:

> The value of the [id](https://www.w3.org/TR/did-core/#dfn-id) property for a [verification method](https://www.w3.org/TR/did-core/#dfn-verification-method) MUST be a [string](https://infra.spec.whatwg.org/#string) that conforms to the rules in Section [3.2 DID URL Syntax](https://www.w3.org/TR/did-core/#did-url-syntax).

There's not really a reason to not follow the did-core spec here right?

Reply:
The did:peer spec also mentions:

> This document should look almost exactly like the final resolved DID Document you desire but with a few key differences:

So I think it is talking about the encoded variant. The resolved DIDDocument should still follow the did-core spec rules I make up of this.